src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -101,28 +101,36 @@ public Resource Merge(Resource other)
                 sanitizedKey = attribute.Key;
             }
 
-            object sanitizedValue;
-            if (!IsValidValue(attribute.Value))
+            object sanitizedValue = SanitizeValue(attribute.Value);
+            if (sanitizedValue == null)
             {
-                OpenTelemetrySdkEventSource.Log.InvalidArgument("Create resource", "attribute value", "Attribute value should be a non-null string, long, bool or double.");
-                sanitizedValue = string.Empty;
-            }
-            else
-            {
-                sanitizedValue = attribute.Value;
+                throw new System.ArgumentException("Attribute value should be a non-null primitive", sanitizedKey);
```

Comment:
this throw could mean either 1. value is null. or 2. value is not null, but not a primitive type. 
Would be nice to throw separate, more explicit message.